pithos/pithos.py

```diff
@@ -161,18 +161,23 @@ def __init__(self, app, test_mode):
         self.set_proxy(reconnect=False)
         self.set_audio_quality()
 
-        email = self.settings['email']
         try:
-            password = get_account_password(email)
+            unlock_keyring()
         except GLib.Error as e:
             if e.code == 2:
-                self.fatal_error_dialog(e.message, _('You need to install a service such as gnome-keyring.'))
+                logging.error('You need to install a service such as gnome-keyring. Error: {}'.format(e))
+                self.fatal_error_dialog(
+                    e.message,
+                    _('You need to install a service such as gnome-keyring.'),
+                )
+
+        email = self.settings['email']
+        password = get_account_password(email)
```

Comment:
I just tested in a Kubuntu 16.04 VM without gnome-keyring installed. It caught the error and showed the fatal dialog. I once we get past the unlocking of the keyring I don't think anything else should raise an exception. We access the keyring when we try to unlock it if anything were to happen it will happen then. But If you want I can include that in the try except block?